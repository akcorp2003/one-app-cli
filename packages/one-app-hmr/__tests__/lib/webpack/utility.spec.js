/*
 * Copyright 2021 American Express Travel Related Services Company, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
 * in compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License
 * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 * or implied. See the License for the specific language governing permissions and limitations
 * under the License.
 */

import {
  createMinifyConfig,
  createOneAppExternals,
  createHotModuleEntries,
} from '../../../lib/webpack/utility';

describe('createMinifyConfig', () => {
  test('returns the optimization webpack config for minifying the output', () => {
    const config = createMinifyConfig();
    expect(config).toMatchSnapshot();
  });

  test('utilizes a cheap-eval-source-map in development mode', () => {
    const config = createMinifyConfig({ isDev: true });
    expect(config.devtool).toEqual('eval-cheap-source-map');
  });
});

describe('createOneAppExternals', () => {
  test('returns the set of externals used by One App', () => {
    const externals = createOneAppExternals();
    expect(externals).toMatchSnapshot();
  });

  test('returns the set of externals used by One App including added externals', () => {
    const externals = createOneAppExternals(['react-package']);
    expect(externals).toMatchSnapshot();
  });

  test('returns the set of externals from added externals in array formation [name, varName]', () => {
    const externals = createOneAppExternals([['react-package', 'reactPackage']]);
    expect(externals).toMatchSnapshot();
  });
});

describe('createHotModuleEntries', () => {
  test('returns an empty object if no modules were supplied', () => {
    const entries = createHotModuleEntries();
    expect(entries).toEqual({});
  });

  test('returns the entries of modules to be bundled', () => {
    const moduleName = 'hot-module';
    const modulePath = '/some/path/to/hot-module';
    const modules = [{ moduleName, modulePath }];
    const entries = createHotModuleEntries(modules);
    expect(entries).toEqual({
      [moduleName]: [
        require.resolve('webpack-hot-middleware/client'),
        require.resolve('react-refresh/runtime'),
        `${modulePath}/src/index.js`,
      ],
    });
  });
});
