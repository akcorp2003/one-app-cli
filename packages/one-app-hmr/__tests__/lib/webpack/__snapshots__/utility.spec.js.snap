// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMinifyConfig returns the optimization webpack config for minifying the output 1`] = `
Object {
  "devtool": false,
  "optimization": Object {
    "minimize": true,
    "minimizer": Array [
      TerserPlugin {
        "options": Object {
          "cache": true,
          "cacheKeys": [Function],
          "exclude": undefined,
          "extractComments": true,
          "include": undefined,
          "minify": undefined,
          "parallel": true,
          "sourceMap": undefined,
          "terserOptions": Object {
            "keep_fnames": true,
          },
          "test": /\\\\\\.jsx\\?\\$/i,
        },
      },
    ],
  },
}
`;

exports[`createOneAppExternals returns the set of externals from added externals in array formation [name, varName] 1`] = `
Object {
  "@americanexpress/one-app-ducks": Object {
    "commonjs2": "@americanexpress/one-app-ducks",
    "root": "OneAppDucks",
    "var": "OneAppDucks",
  },
  "@americanexpress/one-app-router": Object {
    "commonjs2": "@americanexpress/one-app-router",
    "root": "OneAppRouter",
    "var": "OneAppRouter",
  },
  "create-shared-react-context": Object {
    "commonjs2": "create-shared-react-context",
    "root": "CreateSharedReactContext",
    "var": "CreateSharedReactContext",
  },
  "holocron": Object {
    "commonjs2": "holocron",
    "root": "Holocron",
    "var": "Holocron",
  },
  "holocron-module-route": Object {
    "commonjs2": "holocron-module-route",
    "root": "HolocronModuleRoute",
    "var": "HolocronModuleRoute",
  },
  "immutable": Object {
    "commonjs2": "immutable",
    "root": "Immutable",
    "var": "Immutable",
  },
  "prop-types": Object {
    "commonjs2": "prop-types",
    "root": "PropTypes",
    "var": "PropTypes",
  },
  "react": Object {
    "commonjs2": "react",
    "root": "React",
    "var": "React",
  },
  "react-dom": Object {
    "commonjs2": "react-dom",
    "root": "ReactDOM",
    "var": "ReactDOM",
  },
  "react-helmet": Object {
    "commonjs2": "react-helmet",
    "root": "ReactHelmet",
    "var": "ReactHelmet",
  },
  "react-package": Object {
    "commonjs2": "react-package",
    "root": "reactPackage",
    "var": "reactPackage",
  },
  "react-redux": Object {
    "commonjs2": "react-redux",
    "root": "ReactRedux",
    "var": "ReactRedux",
  },
  "redux": Object {
    "commonjs2": "redux",
    "root": "Redux",
    "var": "Redux",
  },
  "reselect": Object {
    "commonjs2": "reselect",
    "root": "Reselect",
    "var": "Reselect",
  },
}
`;

exports[`createOneAppExternals returns the set of externals used by One App 1`] = `
Object {
  "@americanexpress/one-app-ducks": Object {
    "commonjs2": "@americanexpress/one-app-ducks",
    "root": "OneAppDucks",
    "var": "OneAppDucks",
  },
  "@americanexpress/one-app-router": Object {
    "commonjs2": "@americanexpress/one-app-router",
    "root": "OneAppRouter",
    "var": "OneAppRouter",
  },
  "create-shared-react-context": Object {
    "commonjs2": "create-shared-react-context",
    "root": "CreateSharedReactContext",
    "var": "CreateSharedReactContext",
  },
  "holocron": Object {
    "commonjs2": "holocron",
    "root": "Holocron",
    "var": "Holocron",
  },
  "holocron-module-route": Object {
    "commonjs2": "holocron-module-route",
    "root": "HolocronModuleRoute",
    "var": "HolocronModuleRoute",
  },
  "immutable": Object {
    "commonjs2": "immutable",
    "root": "Immutable",
    "var": "Immutable",
  },
  "prop-types": Object {
    "commonjs2": "prop-types",
    "root": "PropTypes",
    "var": "PropTypes",
  },
  "react": Object {
    "commonjs2": "react",
    "root": "React",
    "var": "React",
  },
  "react-dom": Object {
    "commonjs2": "react-dom",
    "root": "ReactDOM",
    "var": "ReactDOM",
  },
  "react-helmet": Object {
    "commonjs2": "react-helmet",
    "root": "ReactHelmet",
    "var": "ReactHelmet",
  },
  "react-redux": Object {
    "commonjs2": "react-redux",
    "root": "ReactRedux",
    "var": "ReactRedux",
  },
  "redux": Object {
    "commonjs2": "redux",
    "root": "Redux",
    "var": "Redux",
  },
  "reselect": Object {
    "commonjs2": "reselect",
    "root": "Reselect",
    "var": "Reselect",
  },
}
`;

exports[`createOneAppExternals returns the set of externals used by One App including added externals 1`] = `
Object {
  "@americanexpress/one-app-ducks": Object {
    "commonjs2": "@americanexpress/one-app-ducks",
    "root": "OneAppDucks",
    "var": "OneAppDucks",
  },
  "@americanexpress/one-app-router": Object {
    "commonjs2": "@americanexpress/one-app-router",
    "root": "OneAppRouter",
    "var": "OneAppRouter",
  },
  "create-shared-react-context": Object {
    "commonjs2": "create-shared-react-context",
    "root": "CreateSharedReactContext",
    "var": "CreateSharedReactContext",
  },
  "holocron": Object {
    "commonjs2": "holocron",
    "root": "Holocron",
    "var": "Holocron",
  },
  "holocron-module-route": Object {
    "commonjs2": "holocron-module-route",
    "root": "HolocronModuleRoute",
    "var": "HolocronModuleRoute",
  },
  "immutable": Object {
    "commonjs2": "immutable",
    "root": "Immutable",
    "var": "Immutable",
  },
  "prop-types": Object {
    "commonjs2": "prop-types",
    "root": "PropTypes",
    "var": "PropTypes",
  },
  "react": Object {
    "commonjs2": "react",
    "root": "React",
    "var": "React",
  },
  "react-dom": Object {
    "commonjs2": "react-dom",
    "root": "ReactDOM",
    "var": "ReactDOM",
  },
  "react-helmet": Object {
    "commonjs2": "react-helmet",
    "root": "ReactHelmet",
    "var": "ReactHelmet",
  },
  "react-package": Object {
    "commonjs2": "react-package",
  },
  "react-redux": Object {
    "commonjs2": "react-redux",
    "root": "ReactRedux",
    "var": "ReactRedux",
  },
  "redux": Object {
    "commonjs2": "redux",
    "root": "Redux",
    "var": "Redux",
  },
  "reselect": Object {
    "commonjs2": "reselect",
    "root": "Reselect",
    "var": "Reselect",
  },
}
`;
